R/nback.R

```diff
@@ -1,41 +1,41 @@
-#' Calculates index scores for N Back games.
+#' N Back Paradigm
 #'
-#' Several values including percentage of correct responses (pc), mean reaction
-#' time (mrt), sensitivity index (i.e, d') and bias (c).
+#' A classical working memory test.
 #'
-#' @param data Raw data of class `data.frame`.
-#' @param ... Other input argument for future expansion.
+#' @templateVar by low
+#' @templateVar vars_input TRUE
+#' @template params-template
 #' @return A [tibble][tibble::tibble-package] contains following values:
 #'   \item{pc}{Percentage of correct responses.}
 #'   \item{mrt}{Mean reaction time.}
 #'   \item{dprime}{Sensitivity index.}
 #'   \item{c}{Bias.}
-#'   \item{is_normal}{Checking result whether the data is normal.}
 #' @export
-nback <- function(data, ...) {
-  vars_output <- c("pc", "mrt", "dprime", "c")
-  vars_required <- tibble::tribble(
-    ~field, ~name,
-    "name_type", "Type",
-    "name_acc", "ACC",
-    "name_rt", "RT"
-  )
-  vars_matched <- match_data_vars(data, vars_required)
-  if (is.null(vars_matched)) {
-    return(compose_abnormal_output(vars_output))
-  }
-  data_cor <- correct_rt_acc(data)
-  basic <- data_cor %>%
-    dplyr::summarise(
-      pc = mean(.data[["acc_cor"]] == 1),
-      mrt = mean(.data[["rt_cor"]], na.rm = TRUE)
+nback <- function(data, by, vars_input) {
+  data_cor <- data %>%
+    dplyr::mutate(
+      # standardize stimuli type
+      type_cor = dplyr::if_else(
+        .data[[vars_input[["name_type"]]]] == "Target",
```

Comment:
N back tests do not use `"Target"` as target name, but use `"Change"`/`"Stay"`. What's more, `"None"` should be ignored.